monai/apps/pathology/metrics.py

```diff
@@ -0,0 +1,212 @@
+
+        # identify isolated tumor cells
+        itc_threshold = (self.itc_diameter + self.grow_distance) / mask_pixel_spacing
+        itc_labels = self.compute_isolated_tumor_cells(tumor_mask=tumor_mask, threshold=itc_threshold)
+
+        return tumor_mask, itc_labels
+
+    def compute_fp_tp(self):
+        """
+        Compute false positive and true positive probabilities for tumor detection,
+        by comparing the model outputs with the prepared ground truths for all samples
+
+        """
+        total_fp_probs, total_tp_probs = [], []
+        total_num_targets = 0
+        num_images = len(self.data)
+
+        for sample in self.data:
+            probs, y_coord, x_coord = self.prepare_inference_result(sample)
+            ground_truth, itc_labels = self.prepare_ground_truth(sample)
+            # compute FP and TP probabilities for a pair of an image and an ground truth mask
+            fp_probs, tp_probs, num_targets = compute_fp_tp_probs(
+                probs=probs,
+                y_coord=y_coord,
+                x_coord=x_coord,
+                evaluation_mask=ground_truth,
+                labels_to_exclude=itc_labels,
+                resolution_level=sample["level"],
+            )
+            total_fp_probs.extend(fp_probs)
+            total_tp_probs.extend(tp_probs)
+            total_num_targets += num_targets
+
+        return (
+            np.array(total_fp_probs),
+            np.array(total_tp_probs),
+            total_num_targets,
+            num_images,
+        )
+
+    def evaluate(self):
+        """
+        Evalaute the detection performance of a model based on the model probability map output,
+        the ground truth tumor mask, and their associated metadata (e.g., pixel_spacing, level)
+        """
+        # compute false positive (FP) and true positive (TP) probabilities for all images
+        fp_probs, tp_probs, num_targets, num_images = self.compute_fp_tp()
+
+        # compute FROC curve given the evaluation of all images
+        fps_per_image, total_sensitivity = compute_froc_curve_data(
+            fp_probs=fp_probs,
+            tp_probs=tp_probs,
+            num_targets=num_targets,
+            num_images=num_images,
+        )
+
+        # compute FROC score give specific evaluation threshold
+        froc_score = compute_froc_score(
+            fps_per_image=fps_per_image,
+            total_sensitivity=total_sensitivity,
+            eval_thresholds=self.eval_thresholds,
+        )
+
+        return froc_score
+
+    def compute_multi_instance_mask(self, mask: np.ndarray, threshold: float):
+        """
+        This method computes the segmentation mask according to the binary tumor mask.
+
+        Args:
+            mask: the binary mask array
+            threshold: the threashold to fill holes
+        """
+        # make sure it is between 0 and 1
+        assert 0 <= mask.max() <= 1, "The input mask should be a binary mask!"
```

Comment:
remove the assert https://deepsource.io/gh/Project-MONAI/MONAI/run/2804f1da-5ea2-4956-940e-a217daca55b5/python/BAN-B101